agent/engine/execcmd/manager_init_task_linux.go

```diff
@@ -150,3 +176,43 @@ func buildContainerNameForBinary(c *apicontainer.Container) string {
 	}
 	return cn
 }
+
+func getExecAgentConfigFileName(sessionLimit int) (string, error) {
+	config := fmt.Sprintf(execAgentConfigTemplate, sessionLimit)
+	hash := getExecAgentConfigHash(config)
+	// check if config file exists already
+	configFilePath := filepath.Join(execAgentConfigDir, fmt.Sprintf(execAgentConfigFile, hash))
+	if execAgentConfigFileExists(configFilePath) {
+		return configFilePath, nil
+	}
+	// config doesn't exist; create a new one
+	if err := createNewExecAgentConfigFile(config, configFilePath); err != nil {
+		return "", err
+	}
+	return configFilePath, nil
+}
+
+func getExecAgentConfigHash(config string) string {
+	hash := sha256.New()
+	hash.Write([]byte(config))
+	sha := base64.URLEncoding.EncodeToString(hash.Sum(nil))
+	return sha
+}
+
+var execAgentConfigFileExists = configFileExists
+
+func configFileExists(configFilePath string) bool {
+	if _, err := os.Stat(configFilePath); err == nil {
+		return true
+	}
+	return false
+}
+
+var createNewExecAgentConfigFile = createNewConfigFile
+
+func createNewConfigFile(config, configFilePath string) error {
+	if err := ioutil.WriteFile(configFilePath, []byte(config), filePerm); err != nil {
```

Comment:
nit: you can just do
```golang
func createNewConfigFile(config, configFilePath string) error {
    return ioutil.WriteFile(configFilePath, []byte(config), filePerm)
}
````

Reply:
ah yes, this is a leftover from a refactoring :)